### PR TITLE
[MM-18980] Migrate components/form-error to TypeScript

### PR DESCRIPTION
**Pull request by kgeorgiou**
#### Summary

This migrates `components/form_error` to TypeScript.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/12448

#### Related Pull Requests
N/A

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -5455,6 +5455,7 @@
           "resolved": "https://registry.npmjs.org/p-timeout/-/p-timeout-2.0.1.tgz",
           "integrity": "sha512-88em58dDVB/KzPEx1X0N3LwFfYZPyDc4B6eF38M1rk9VTZMbxXXgjugz8mmwpS9Ox4BDZ+t6t3QP5+/gazweIA==",
           "dev": true,
+          "optional": true,
```
Comment (saturninoabril) [inline package-lock.json]: Please revert all changes in this file.

File: components/form_error.tsx [hunk 2]
```diff
@@ -1,39 +1,35 @@
...
             return null;
         }
 
         // look for the first truthy error to display
-        let message = this.props.error;
+        let message = error;
 
         if (!message) {
-            for (const error of this.props.errors) {
+            for (const error of errors) {
```
Comment (saturninoabril) [inline components/form_error.tsx]: There are some style errors.  Please run ``npm run check`` and fix accordingly.

**Comment by saturninoabril** [comment]
@kgeorgiou Please update to master so we can merge, thanks

**Comment by kgeorgiou** [comment]
@saturninoabril I think @devinbinnie merged master into this earlier today. Should I still rebase on top of master or is it good to go? 🚀

**Comment by saturninoabril** [comment]
Thanks @kgeorgiou, that's all good!

**Pull request merged**